src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -696,8 +707,41 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
     template<typename T, typename Converter>
     void SetValueForKey(Json::Value& json, std::string_view key, const T& target, Converter&& conv)
     {
-        // demand guarantees that it will return a value or throw an exception
-        *json.demand(&*key.cbegin(), (&*key.cbegin()) + key.size()) = conv.ToJson(target);
+        std::optional<Json::Value> out;

```

Comment:
i also don't 100% understand why this had to be an optional instead of you caching the pointer returned by `demand`...

Reply:
The second you use `demand`, you are writing out to the Json::Value. Even if you don't pass anything in, it'll at the very least write out `null` because it assumes the json key exists. In our json model, `null` has meaning (jsoncpp lets you determine if you want null to be written out or ignored). So we only want to do a `demand` call when we know we actually want to write a value out.

Adding a similar comment in the code.

Reply:
Oh wait, I already have that comment on line 739 haha